### ISSUE TITLE
Provide log deconstruction mechanism that leverages file names

### ISSUE THREAD

**Issue by michaelcfanning** [oasis-tcs/sarif-spec#211]
Consider a file named MyOutput.sarif. On opening this file, a consumer could search for MyOutput.files.sarif in order to locate a 'files' table that is associated with MyOutput.sarif. 

This approach would prevent the need to have SARIF properties have two types (either a URI or the property itself). We might to consider a mechanism to allow retrieving this content from the web.

Merging this idea with #210, we could also simply consider allowing the 'master' file to include a set of URLs to consult for populating the data rather than inlining the URLs:

indirectSarifDataSources
{
"files" : [ { "file:///e:/MySarif.files.sarif"}
}

**Comment by lgolding** [comment]
Yes, as we discussed on the phone: We would search for an element such as "run.files" in the following order:

1. If the element is present in the SARIF file, use it.
    OTHERWISE
2. If the SARIF file specifies a URL (or an array of URLs, if we decide to allow that) for the element, get the element from there.
    OTHERWISE
3. If there is a resource available at a conventionally named URL, use that.
    OTHERWISE
4. The element is missing -- and it the element is _required_, that's an error.

**Comment by lgolding** [comment]
@michaelcfanning As we discussed on the phone, this "configuration by convention" is an unnecessary complication. It also burdens consumers to probe for a conventionally named file -- which frequently won't be there -- whenever an optional, "externalizable" property is missing from a log file.

Without the configuration by convention feature, the lookup is the same as in my comment above, but without Step 3.

Closing this issue.
